src/nodes.coffee

```diff
@@ -1077,7 +1077,7 @@ exports.Value = class Value extends Base
   isBoolean      : -> @bareLiteral(BooleanLiteral)
   isAtomic       : ->
     for node in @properties.concat @base
-      return no if node.soak or node instanceof Call
+      return no if node.soak or node instanceof Call or node instanceof Op and node.operator is 'do'
```

Comment:
This had to be added for `Parens` not to decide to unparenthesize its output for eg `new (do ->).b`

I didn't totally grasp conceptually what `isAtomic()` means but this seems to make sense as `do` is semantically equivalent to a `Call`